execution/executor/src/lib.rs

```diff
@@ -977,6 +978,7 @@ impl<V: VMExecutor> BlockExecutor for Executor<V> {
             block.output().executed_trees().state_tree().prune()
         }
 
+        // old root is pruned here
         write_lock.prune(ledger_info_with_sigs.ledger_info())?;
```

Comment:
I THINK, with the postponed pruning, this entire part can be skipped. Since with all previous blocks deallocated, unlinking the `base` field from the latest root will cause all previously linked nodes' ref counts drop to 0 hence properly freed.